### DISCUSSION TITLE
Cron bash script virtnbdbackup | mail only if error?

### DISCUSSION THREAD

**Discussion by slu-010101** [abbbi/virtnbdbackup#171] [General]
Hi,

do anyone use virtnbdbackup with a cronjob?
I start virtnbdbackup in my bash script, but every start produce a long and "unreadable" cron mail:

> [...]
> 54%|█████▎ | 7.40G/13.8G [00:12<00:09, 690MB/s] saving disk vda: 
> 54%|█████▍ | 7.50G/13.8G [00:12<00:08, 698MB/s]
> saving disk vda: 55%|█████▌ | 7.58G/13.8G [00:12<00:08, 695MB/s] saving disk vda: 
> 56%|█████▌ | 7.67G/13.8G [00:12<00:09, 656MB/s] saving disk vda: 
> 56%|█████▋ | 7.77G/13.8G [00:12<00:09, 660MB/s] saving disk vda: 
> 57%|█████▋ | 7.87G/13.8G [00:13<00:08, 661MB/s] saving disk vda: 
> 58%|█████▊ | 7.95G/13.8G [00:13<00:08, 687MB/s] saving disk vda: 
> 58%|█████▊ | 8.03G/13.8G [00:13<00:08, 712MB/s] saving disk vda: 
> [...]

Version 2.4 with info log level, I need only a mail from cron if something is wrong/failed.
Or some short overview, backup type (full,inc,...) and amount of data in the backup...

**Comment by abbbi** [comment]
you can disable progressbar with --noprogress or use --quiet, it disables logging output but will exit with error code accordingly, as such you can script around to send logfile if backup fails.

**Comment by slu-010101** [comment]
Thanks for the hint, --noprogress make sense.

--quiet is much so quiet for me, I like to see some details* for example:
```
name of VM
backup type (full,inc,...)
amount of backup data
required backup time
```

*This is more an enhancement, I know.

**Comment by billystaggs** [comment]
I just getting started but I am running via crontab and have it setup to email on errors. Here are the relevant lines.  Amateur work, but it's working fine: 

errstate="false"

virtnbdbackup -d Gibson -l auto -o $bakdir &>> $templog

cat $templog >> $adminlog
if grep -qi 'error' $templog; then
        errstate='true'
fi

# if error say so
if $errstate = "true"; then
        echo 'Status: ERROR! ' >> $adminlog
else
        echo 'Status: Successful ' >> $adminlog
fi

# mail error to sendto
if $errstate = "true"; then
        cat $adminlog | mailx -s "VM Backup Error!" $sendto
else
        cat $adminlog | mailx -s "VM Backup Report" $sendto
fi

--
bstaggs

**Comment by aroskamp** [comment]
You might want to take a look at [Cronic](https://github.com/justincase/cronic)

I worked with it for some other projects. Cronic only sends emails, when the internal script writes to stderr or when exitcode != 0.

**Comment by slu-010101** [comment]
Send a mail only if there is an error is not enough, I need the following information:
 VM name
 Total saved disk data
 Finished successfully

At the moment I received a email every night from the cron job.
My idea is now to write the cron output into a logfile and grep than for this information.

Or do someone has an better idea?
